lib/public/views/Runs/RunPerDataPass/RunsPerDataPassOverviewModel.js

```diff
@@ -99,6 +104,49 @@ export class RunsPerDataPassOverviewModel extends FixedPdpBeamTypeRunsOverviewMo
         }
     }
 
+    /**
+     * Fetch skimmable runs for given data pass
+     * @return {Promise<void>} resolves once data are fetched
+     */
+    async _fetchSkimmableRuns() {
+        this._skimmableRuns$.setCurrent(RemoteData.loading());
+        try {
+            const { data: skimmableRuns } = await getRemoteData(buildUrl('/api/dataPasses/skimming/runs', { dataPassId: this._dataPassId }));
+            const runToReadyForSkimmingFlag = Object.fromEntries(skimmableRuns
+                .map(({ runNumber, readyForSkimming }) => [runNumber, readyForSkimming]));
+            this._skimmableRuns$.setCurrent(RemoteData.success(runToReadyForSkimmingFlag));
+        } catch (error) {
+            this._skimmableRuns$.setCurrent(RemoteData.failure(error));
+        }
+    }
+
+    /**
+     * Update skimmable runs for given data pass
+     *
+     * @param {number} runNumber run number
+     * @return {Promise<void>} resolves once data are fetched
+     */
+    async updateSkimmableRuns(runNumber) {
+        this._skimmableRuns$.getCurrent().match({
+            Success: async (currentSkimmableRuns) => {
+                this._skimmableRuns$.setCurrent(RemoteData.loading());
```

Comment:
This seems very strange to update the remote data in a match. Why do you need it to be success?

Reply:
To update skimmable runs, first, you need to successfully fetch current state.
Added throwing an error when the method is called having `skimmableRuns$` in state different than Success